src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/PrivateDataRetrievalService.java

```diff
@@ -103,13 +112,16 @@ private PrivateBoDataListApi getBeneficialOwnersData(String companyNumber)
         ApiLogger
             .info(String.format("Retrieved %d Beneficial Owners for Company Number %s",
                 numberOfBOs, companyNumber));
-      } else {
-        ApiLogger.info("No Beneficial Owners found for Company Number " + companyNumber);
       }
 
       return beneficialOwnerDataList;
-
-    } catch (URIValidationException | IOException e) {
+    } catch (ApiErrorResponseException e) {
+      if (e.getStatusCode() == HttpServletResponse.SC_NOT_FOUND) {
+        ApiLogger.errorContext("No Beneficial Owners found for Company Number "+companyNumber, e);
```

Comment:
please use ApiLogger.error(message, e, logMap); like line 67 so that we can search logs for company number key.